### ISSUE TITLE
Deploy command doesn't copy devDependencies

### ISSUE THREAD

**Issue by ariesclark** [pnpm/pnpm#4992]
### Code to reproduce the issue:
```sh
# Inside apps/api/dist, run the following command.
pnpm deploy --filter \".\" apps/api/dist
```

### Expected behavior:
Deploy would copy all dependencies, including ``devDependencies``.

### Actual behavior:
Only packages inside ``dependencies`` are included in the output ``node_modules``.

### Additional information:

 - `node -v` prints: v18.2.0
 - Windows, macOS, or Linux?: Ubuntu 22.04 LTS x86_64

**Comment by ariesclark** [comment]
Also, the resulting directory doesn't contain a ``pnpm-lock.yaml``, but that's not relevant to this particular issue.

**Comment by robations** [comment]
I've not tried this feature, but I would have guessed that devDependencies are only needed for development, not to be deployed?

**Comment by ariesclark** [comment]
fixed in #5040
